File: analytical-web/src/app/models/Questions.ts
import { QueryDocumentSnapshot } from '@angular/fire/firestore';

export interface Questions {
  id: string;
  gameID: string;
  levelID: string;
  question: string;
  image: string | null;
  hint: string;
  answer: string;
  choices: string[];
  type: 'QUIZ_GAME' | 'PUZZLE_GAME' | 'MEMORY_GAME' | 'MATH_GAME';
  createdAt: Date;
  updatedAt: Date;
}

export const questionConverter = {
  toFirestore: (data: Questions) => data,
  fromFirestore: (snap: QueryDocumentSnapshot) => {
    const question = snap.data() as Questions;
    question.createdAt = (question.createdAt as any).toDate();
    question.updatedAt = (question.updatedAt as any).toDate();
    return question;
  },
};

File: Analytical/app/src/main/java/com/ketchupzzz/analytical/models/questions/Questions.kt
package com.ketchupzzz.analytical.models.questions

import android.os.Parcelable
import com.ketchupzzz.analytical.models.Category
import java.util.Date


@kotlinx.parcelize.Parcelize
data class Questions(
    val id : String ? = null,
    val gameID : String ? = null,
    val levelID : String ? = null,
    val image  : String ? = null,
    val question: String ? = null,
    val hint : String ? = null,
    val type : Category? = null,
    val choices : List<String> = mutableListOf(),
    val answer: String ? = null,
    val createdAt : Date = Date(),
    val updatedAt : Date = Date()
): Parcelable


